### ISSUE TITLE
socks5 to specific interfaces

### ISSUE THREAD

**Issue by incirrata** [b23r0/rsocx#12]
Apologies for in-formalities. But I am looking to develop with this code. I would like to modify the the sock.rs to push to different routes or interfaces or connections in a round robin or random fashion. I understand how to do the round robin part, but I am very unfamiliar with rust. Was hoping if you knew off the top of your head any specific functions I could user to accomplish this goal. Thanks 
Something like:
```
use tokio::net::TcpSocket;

use std::io;

#[tokio::main]

async fn main() -> io::Result<()> {

    let addr = "127.0.0.1:8080".parse().unwrap();
    
    let socket = TcpSocket::new_v4()?;
    let stream = socket.connect(addr).await?;

    Ok(())
}
```

**Comment by b23r0** [comment]
Hey bro, you dont need to apologize at all, you can freedom to modify and use the code of this project under the MIT license.

But I think you may need to learn how to program with rust a little further. I guess your needs may require some modifications in socks.rs::tcp_transfer(), such as choosing a specific interface before establishing a connection with the remote.

Since I dont know your real intentions, I cant give you more advice.

**Comment by incirrata** [comment]
Ahh No problem.
I've started learning Rust yesterday from commenting out your code and learning some of the concepts work for Rust. I found tokio has a means of selecting an IP that I could incorporate into a pub enum declaration that grabs specific interface IP at runtime. Then modify the TCP tx() to divert through their connections via socks5 proxy and some routing magic. 

At the end of the day my goal is to use rsocx as a means to divert traffic to multiple VPN endpoints from one proxy connection via client HTTP request or TCP Stream identification.

So for instance, a server has 20x different Wireguard vpn connections all connected at once. 
Modifying rsocx's fn async function that sees the TCP stream will coordinate, or maybe have logic to migrate tcp_tx() to a different interface/IP/route upon a new TCP_stream or HTTP_stream. (Me guessing the way to do it)

So at the end of the day, if I ran curl --socks5 dest_ip:8443 -k https://whatsmyip.yadda yadda
 
It would produce a random different remote IP address each time I run it coordinating to a different location 

Again excuse my newbie speak lol
